sql-plugin/src/main/330+/scala/org/apache/spark/sql/rapids/shims/intervalExpressions.scala

```diff
@@ -0,0 +1,497 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.shims
+
+import ai.rapids.cudf.{BinaryOperable, ColumnVector, DType, RoundMode, Scalar}
+import com.nvidia.spark.rapids.{Arm, GpuBinaryExpression, GpuColumnVector, GpuScalar}
+
+import org.apache.spark.sql.catalyst.expressions.{Expression, ImplicitCastInputTypes, NullIntolerant}
+import org.apache.spark.sql.types._
+
+object IntervalUtils extends Arm {
+
+  /**
+   * compute interval.toLong * num
+   */
+  def mulIntervalAndNum(interval: BinaryOperable, num: BinaryOperable): ColumnVector = {
```

Comment:
I am not sure, but did you try
```
interval.mul(num, INT64)
```
instead ?

Reply:
Updated, good idea.